sympy/core/assumptions.py

```diff
@@ -621,6 +623,16 @@ def __init__(cls, *args, **kws):
             if eval_is_meth is not None:
                 cls._prop_handler[k] = eval_is_meth
 
+            # When __init__ is called from UndefinedFunction it is called with
+            # just one arg. When it is called from subclassing Basic it is
+            # called with the usual (name, bases, namespace) type() signature.
+            if len(args) == 3:
+                namespace = args[2]
+                is_assump = 'is_%s' % k
+                if is_assump in namespace:
+                    if isinstance(namespace[is_assump], property):
+                        warnings.warn(f"{is_assump} is defined as a property method on {cls}. Doing so will break the automatic deduction of other assumptions. Either define {is_assump} as a simple bool or define _eval_{is_assump} instead.", stacklevel=2)
```

Comment:
Just to clear up the codebase this should be turned into an error until CI passes.